### ISSUE TITLE
Reduce cardinality in Dapr metrics and add more information to API logs

### ISSUE THREAD

**Issue by ItalyPaleAle** [dapr/dapr#6723]
## In what area(s)?

/area runtime

## What version of Dapr?

1.1.x
master

## Description

Related: #6581

The Dapr HTTP server collects metrics with very high cardinality by default. Each path invoked in the HTTP server is used to create a "bucket" in the metrics, and this causes significant memory usage when using Dapr to invoke RESTful APIs (where the path is usually different on every request).

This happens in service invocation metrics as well as in the general HTTP server metrics.

Users can manually configure Dapr to [reduce the cardinality of metrics,](https://docs.dapr.io/operations/monitoring/metrics/metrics-overview/#high-cardinality-metrics) but that is a manual process that is not immediately clear, and it requires writing regular expressions.

@yaron2 and I have been discussing how to address this issue and our proposal is:

1. By default, do not report metrics per each endpoint/path in the HTTP server. Aggregation will stop at the app ID for service invocation.
2. Add an option to the Configuration CRD `metrics.highCardinality`, disabled by default, to retain the current behavior.
3. Change the API logging middleware to include additional information on each request, including latency and returned status code. These are things that were before included in the metrics. Although API logging aren't a 1:1 replacement for metrics, users can leverage them in tools like Grafana to gather similar insights.

## Release Note

RELEASE NOTE: **CHANGED** Reduced default cardinality of metrics collected by HTTP server

**Comment by Hsuwen** [comment]
About this issue, I found that the docs has been updated.
[https://docs.dapr.io/operations/monitoring/metrics/metrics-overview/#high-cardinality-metrics](url)

Several details are still not very clear:
1. What specific metrics will be affected by this will determine which metrics I should set
2. How to filter by appid in the configuration of ```metric.rules``` 

Regarding point **2**:
If there are paths like ```/users/<id>``` in both app1 and app2, but only app1 needs to be set, how can we do this? Of course, it is possible to configure each app separately, but this will increase complexity.
What I mean is, can we do this:
```
  metric:
    enabled: false
    rules:
    - name: dapr_runtime_service_invocation_req_sent_total
      labels:
      - name: method
        appid: app1
        regex:
          "users/": "users/.+"
```
